src/scm/gitlab.js

```diff
@@ -0,0 +1,278 @@
+
+class Gitlab {
+  constructor(baseUrl, user, token) {
+    this.baseUrl = baseUrl;
+    this.user = user;
+    this.namesToIds = {
+      repos: {},
+      groups: {}
+    };
+    this.accessToken = token.token;
+    if(token.type === 'oAuth') {
+      this.tokenParam = 'access_token=' & this.accessToken;
+      this.tokenHeader = {
+        'Authorization': `Bearer ${this.accessToken}`
+      };
+    } else {
+      this.tokenParam = 'private_token=' & this.accessToken;
+      this.tokenHeader = {
+        'Private-Token': this.accessToken
+      };
+    }
+    this.namespaces = [user];
+  }
+
+  get name() {
+    return 'gitlab';
+  }
+
+  get canUseGist() {
+    return false;
+  }
+
+
+  commitFiles(repo, branch, parent, newFiles, changedFiles, deleteFiles, comment) {
+    return new Promise((resolve, reject) => {
+      let data = {branch: branch,
+        commit_message: comment,
+      actions: []};
+      if (newFiles && newFiles.length > 0) {
+        data.actions = data.actions.concat(newFiles.map((file) => {
+          return {
+            action: 'create',
+            file_path: file.name,
+            content: file.content
+          }
+        }));
+      }
+      if (changedFiles && changedFiles.length > 0) {
+        data.actions = data.actions.concat(changedFiles.map((file) => {
+          return {
+            action: 'update',
+            file_path: file.name,
+            content: file.content
+          }
+        }));
+      }
+      if (deleteFiles && deleteFiles.length > 0) {
+        data.actions = data.actions.concat(deleteFiles.map((file) => {return {
+          action : 'delete',
+          file_path : file
+        }}));
+      }
+      $.ajax({
+        url: `${this.baseUrl}/projects/${context.repo.id}/repository/commits`,
+        headers: this.tokenHeader,
+        contentType: 'application/json',
+        method: 'POST',
+        crossDomain: true,
+        traditional: true,
+        data: JSON.stringify(data)
+      })
+        .then(resolve)
+        .fail(reject);
+    });
+  }
+
+  push(code) {
+    const changed = $('.diff-file:checked').toArray().map(elem => elem.value);
+    const changedFiles = changed.filter(f => code.gas[f]).map(f => {
+      return {name: f.replace(/\.gs$/, context.config.filetype), content: code.gas[f]}
+    });
+    const deleteFiles = changed.filter(f => !code.gas[f]);
+    const newFileNames = changed.filter(f => !code.scm[f]);
+    const updatedFileNames = changed.filter(f => !newFileNames.includes(f));
+
+    const newFiles = changedFiles.filter(f => newFileNames.includes(f.name));
+    const updatedFiles = changedFiles.filter(f => updatedFileNames.includes(f.name));
+
+    const comment = $('#commit-comment').val();
+
+    this.commitFiles(context.repo.fullName, context.branch, null, newFiles, updatedFiles, deleteFiles, comment)
+      .then(() => {
+        showAlert(`Successfully push to ${context.branch} of ${context.repo.fullName}`);
+      })
+      .catch((err) => {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/5becb274ac17be98fee6e945e23cbab998e9d302/src/scm/gitlab.js#L96"></span>‘err’ is defined but never used.</td>
    </tr>
  </tbody>
</table>